### PR TITLE
Faster summarize by gene

### PR DESCRIPTION
**Pull request by hfuchs5**
**What changes are proposed in this pull request?**
Avoid use of pivot_*() functions in this because it was taking too long. I used base R matrices instead. Now it takes 10 seconds to run according to `tictoc` package.

**If there is an GitHub issue associated with this pull request, please provide link.**
#259 

--------------------------------------------------------------------------------

  Reviewer Checklist (if item does not apply, mark is as complete)

- [x] PR branch has pulled the most recent updates from main branch. Ensure the pull request branch and your local version match and both have the latest updates from the main branch.
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded.
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
 - [x] Update `NEWS.md` with the changes from this pull request under the heading "`# cbioportalR (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] Run `codemetar::write_codemeta()`
- [x] Run `usethis::use_spell_check()` again
- [x] Approve Pull Request
- [x] Merge the PR

### CODE REVIEW

**Comment by michaelcurry1123** [comment]
in `summarize_by_gene()` on line 35 when you name the columns it is possible that they have the same name. After this you cannot use mutate or other dplyr functions. If you are trying to name the columns after the sample_id you can maybe do something like this: `all_bin2[rownames(all_bin2)== "sample_id",]`

File: R/summarize-by-gene.R [hunk 1]
```diff
@@ -25,31 +25,104 @@ summarize_by_gene <- function(gene_binary) {
...
+    summarise(count = n())%>%
+    filter(count > 1)
+
+  genes_multiple <- test$name2
+
+  # genes only with one type of event
+  all_bin_once <- all_bin2 %>%
+    filter(!(name2 %in% genes_multiple))%>%
+    select(-gene)%>%
+    mutate(across(starts_with("P"), as.numeric))
```
Comment (karissawhiting) [inline R/summarize-by-gene.R]: this may not generalized to non IMPACT sample IDs

File: R/summarize-by-gene.R [hunk 2]
```diff
@@ -25,31 +25,104 @@ summarize_by_gene <- function(gene_binary) {
...
+
+  row.names(all_bin_once) <- NULL
+
+  # genes with more than one type of event
+  all_bin_more <- all_bin2 %>%
+    filter(name2 %in% genes_multiple)%>%
+    mutate(across(starts_with("P"), as.numeric))%>%
+    select(-gene)%>%
+    group_by(name2)%>%
+    summarize(across(everything(), max))
```
Comment (karissawhiting) [inline R/summarize-by-gene.R:79]: write tests for cases with all NA, all 0, all 1

File: R/summarize-by-gene.R [hunk 3]
```diff
@@ -25,31 +25,104 @@ summarize_by_gene <- function(gene_binary) {
     cli::cli_abort("{.code gene_binary} must be a data.frame with sample ids")
   }
 
+  defaultW <- getOption("warn")
+  options(warn = -1)
```
Comment (karissawhiting) [inline R/summarize-by-gene.R]: Unfortunately I think CRAN may give us issues if we change global options here

**Comment by karissawhiting** [comment]
@hfuchs5 I made some updates to simplify how to handle `sample_id` (you're right that rownames can't be avoided unfortunately!) and add a few speed enhancements (e.g. using `table()` instead of `summarise`). I also added a check for numeric columns up front instead of transforming later which seems to speed things even more!

```
        test replications                elapsed            relative   user.self   sys.self
1      summarize_by_gene_new           500 161.339     1.000   156.353      5.092
2      summarize_by_gene               500 229.061    1.420   222.734      6.407
```

Could you test it out with your large data set to make sure it still is faster?

Thanks for your work on this. I think we are close!

**Comment by hfuchs5** [comment]
It ran through my dataset with 1000+ samples in 1.14 seconds! It looks good to me and I'm going to merge.

**Pull request merged**